app/Http/Controllers/CategoriesController.php

```diff
@@ -112,23 +105,31 @@ public function edit($categoryId = null)
      * @see CategoriesController::getEdit() method that makes the form.
      * @param ImageUploadRequest $request
      * @param int $categoryId
-     * @return \Illuminate\Http\RedirectResponse
-     * @throws \Illuminate\Auth\Access\AuthorizationException
      * @since [v1.0]
      */
-    public function update(ImageUploadRequest $request, $categoryId = null)
+    public function update(ImageUploadRequest $request, $categoryId = null) : RedirectResponse
     {
         $this->authorize('update', Category::class);
         if (is_null($category = Category::find($categoryId))) {
             // Redirect to the categories management page
-            return redirect()->to('admin/categories')->with('error', trans('admin/categories/message.does_not_exist'));
+            return redirect()->route('categories.index')->with('error', trans('admin/categories/message.does_not_exist'));
         }
 
         // Update the category data
         $category->name = $request->input('name');
         // If the item count is > 0, we disable the category type in the edit. Disabled items
         // don't POST, so if the category_type is blank we just set it to the default.
+
+
+        // Don't allow the user to change the category_type once it's been created
+        if (($request->filled('category_type') && ($category->itemCount() > 0))) {
```

Comment:
This is a worthy, and valid, and useful set of changes to make - but jamming this into a completely unrelated PR is gauche. I'll still take it.

Reply:
We had no tests for this behavior, and the only way to write tests and actually have them pass was this.